### ISSUE TITLE
More description for optimizers Adagrad, Adadelta, FTRL...

### ISSUE THREAD

**Issue by kemaswill** [tensorflow/tensorflow#9731]
Is it necessary to add more description for optimizers such as Adagrad, Adadelta, FTRL and so on just as what [Adam](https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer) does? Since these are several quite new optimizers and I think it's better to show users more details about these optimizers so that they can understand why do these optimizers work better than SGD in some situations.

If more descriptions are welcomed, I'm glad to make new PRs to do this.

Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by kemaswill** [comment]
@wolffg @aselle

**Comment by girving** [comment]
What kind of extra language would you want?

**Comment by kemaswill** [comment]
Just English, but to show more details of these optimization algorithms, such as how are the parameters updated, how do these methods adapt their learning rates.

**Comment by girving** [comment]
@wolffg I'll let you decide how much we want to do here.

**Comment by amejs** [comment]
Hello, 
I have a question about "AdagradOptimizer", I created a node for the learning rates to be updated but I don't see any change from the initial learning rate. 

This is the code for the optimizer within the Trainer object that I'm using

self.learning_rate_node = tf.Variable(learning_rate)
optimizer = tf.train.AdagradOptimizer(learning_rate=learning_rate).minimize(self.net.cost, global_step=global_step)

Do I have to specify something else in "AdagradOptimizer" for the learning rate to be updated? Do I have to define the decay that I want to use when I create "self.learning_rate_node"? Thanks in advance.

Best regards, Amelia.

**Comment by tensorflowbutler** [comment]
This issue is automatically closed due to lack of activity. Please re-open if this is still an issue for you. Thanks!
